### PR TITLE
ramfs and installcommand: support external commands

### PR DESCRIPTION
**Pull request by rminnich**
External commands like /src/github.com/elves/elvish
don't follow standard rules.

This change allows finding the source if it's not
in the standard place. I also took the opportunity to
tidy things up a bit and also remove what looked like an error,
namely if the command was found but noforce was set, we'd do
nothing, which is not correct.

For testing the build on darwin, I also moved some things
around in uroot/. These are not seperate out but I hope
that's ok.

But I can now do this:
go run scripts/ramfs.go -test 'src/github.com/u-root/u-root/cmds/[a-z]*' src/github.com/elves/elvish

And I get the elvish shell if I just type
elvish

Signed-off-by: Ron Minnich <rminnich@gmail.com>

### CODE REVIEW

File: cmds/init/init.go [hunk 1]
```diff
@@ -49,29 +49,34 @@ func main() {
...
-					log.Printf("Symlink %v -> %v failed; %v", source, destPath, err)
-				}
-				if *ludicrous {
-					log.Printf("Symlink %v -> %v", source, destPath)
-				}
+	c, err := filepath.Glob("/src/github.com/u-root/u-root/cmds/[a-z]*")
+	if err != nil || len(c) == 0 {
+		log.Printf("In a break with tradition, you seem to have NO u-root commands: %v", err)
+	}
+	o, err := filepath.Glob("/src/*/*/*")
```
Comment (hugelgupf) [inline cmds/init/init.go]: this seems fragile. what if this stuff isn't coming from github or is at a lower level in a github dir? packages can be at any level below /src in go.

File: cmds/installcommand/installcommand.go [hunk 2]
```diff
@@ -43,7 +43,7 @@ var (
 	urpath    = "/go/bin:/ubin:/buildbin:/usr/local/bin:"
 	lowpri    = flag.Bool("lowpri", false, "the scheduler priority is lowered before starting")
 	ludicrous = flag.Bool("ludicrous", false, "print out ALL the output from the go build commands")
-	noexec    = flag.Bool("noexec", false, "do not execute the process after building")
+	build     = flag.Bool("build", false, "bulid, i.e. do not execute the process after building")
```
Comment (hugelgupf) [inline cmds/installcommand/installcommand.go]: *build

maybe say "only build"

File: uroot/uname_darwin.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// Copyright 2015-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// package uroot contains various functions that might be needed more than
```
Comment (hugelgupf) [inline uroot/uname_darwin.go]: no need for package comment since it's already in other files.

File: cmds/init/init.go (hunk 1 above)
Comment (rminnich) [inline cmds/init/init.go]: It is fragile. I'm just not sure what to do about it. How should we walk source and how do we figure out we've hit a command? I've no idea.

File: cmds/installcommand/installcommand.go (hunk 2 above)
Comment (rminnich) [inline cmds/installcommand/installcommand.go]: yeah. I like onlybuild. Or buildonly?

File: uroot/uname_darwin.go (hunk 3 above)
Comment (rminnich) [inline uroot/uname_darwin.go]: done

File: cmds/init/init.go (hunk 1 above)
Comment (rminnich) [inline cmds/init/init.go]: although if it's not coming from github, this still works. I think the common convention for commands is mostly workable here. We can fix exceptions should they ever occur.

Comment (xchenan) [inline cmds/init/init.go]: @hugelgupf The changes look good to me. I've tried both on chroot and qemu. The codes works fine with starting up elvish. 

For the /src/*/*/* glob, Ron is willing to accept that for now, as one of the u-root rules.

**Pull request merged**